packages/bot/src/stores/flyout-store.js

```diff
@@ -212,6 +216,34 @@ export default class FlyoutStore {
         this.is_help_content = true;
     }
 
+    /**
+     * Close the flyout on click outside itself or parent toolbox.
+     */
+    @action.bound
+    clickOutsideFlyoutCallback(event) {
+        if (!this.is_visible || !Blockly.derivWorkspace) {
+            return;
+        }
+
+        const toolbox         = Blockly.derivWorkspace.toolbox_; // eslint-disable-line
+        const is_flyout_click = event.path.some(el => el.classList && el.classList.contains('flyout'));
+        const isToolboxClick  = () => toolbox.HtmlDiv.contains(event.target);
+
+        if (!is_flyout_click && !isToolboxClick()) {
+            toolbox.clearSelection();
+        }
+    }
+
+    @action.bound
+    onMount() {
+        window.addEventListener('click', this.clickOutsideFlyoutCallback);
```

Comment:
@aaron-binary this callback function can go to scratch-helper.  btw flyout is already being closed when we click outside. why you need to change it?
and it's better if we could have and event like `workspace.click` or something like that that we can know when we click on workspace and then do anything we need there?